vllm/model_executor/layers/quantization/fp_eXmY.py

```diff
@@ -0,0 +1,208 @@
+                         prefix: str) -> Optional["FP_eXmYLinearMethod"]:
+        if isinstance(layer, LinearBase):
+            return FP_eXmYLinearMethod(self)
+        return None
+
+
+class FP_eXmYLinearMethod(LinearMethodBase):
+    """Linear method for FP_eXmY quantizer.
+    Args:
+        quant_config: the FP_eXmY quantization config.
+    """
+
+    def __init__(self, quant_config: FP_eXmYConfig):
+        self.quant_config = quant_config
+        self.weight = None
+
+    def create_weights(self,
+                       layer: torch.nn.Module,
+                       input_size_per_partition: int,
+                       output_partition_sizes: List[int],
+                       input_size: int,
+                       output_size: int,
+                       params_dtype: torch.dtype,
+                       weight_loader=None,
+                       **extra_weight_attrs):
+        del output_size
+        del input_size
+        output_size_per_partition = sum(output_partition_sizes)
+        weight = FP_eXmYParameter(
+            torch.Size((output_size_per_partition, input_size_per_partition)),
+            params_dtype=params_dtype,
+            quant_config=self.quant_config,
+        )
+        set_weight_attrs(weight, {
+            "input_dim": 1,
+            "output_dim": 0,
+        })
+        layer.register_parameter("weight", weight)
+
+        def quant_weight_loader(param, loaded_weight, *args, **kwargs):
+            # Calls the original weight loader (if any), quantizes the result,
+            # and then loads the quantized parameter.
+            if weight_loader is not None:
+                orig_param_data = param.data
+                param.data = param.quant_llmdequantize()
+                weight_loader(param, loaded_weight, *args, **kwargs)
+                param.data, loaded_weight = orig_param_data, param.data
+            param.quant_llmquantize_(loaded_weight.cuda())
+
+        extra_weight_attrs["weight_loader"] = quant_weight_loader
+        set_weight_attrs(weight, extra_weight_attrs)
+
+    def apply(self,
+              layer,
+              x: torch.Tensor,
+              bias: Optional[torch.Tensor] = None) -> torch.Tensor:
+        weight = layer.weight
+        weights = weight.data
+        scales = weight.scales
+        out_dim, in_dim = weights.shape
+        bsize = x.reshape(-1, x.shape[-1]).shape[0]
+        splitK = _SPLIT_K_MAP[(bsize - 1) //
+                              64].get(out_dim, 1) if bsize <= 768 else 1
+        if bias is None:
+            return ops.fp_eXmY_linear_forward_cuda(
+                self.quant_config.exponent_bits,
+                self.quant_config.mantissa_bits, x, weights, scales, splitK)
+        else:
+            return ops.fp_eXmY_linear_forward_cuda(
+                self.quant_config.exponent_bits,
+                self.quant_config.mantissa_bits, x, weights, scales,
+                splitK) + bias
```

Comment:
```suggestion
        reshape_after_matmul = False
        if x.ndim > 2:
            original_shape = x.shape
            x = x.reshape(-1, x.size(-1))
            reshape_after_matmul = True

        out = ops.fp_eXmY_linear_forward_cuda(
                self.quant_config.exponent_bits,
                self.quant_config.mantissa_bits, x, weights, scales, splitK)

        if reshape_after_matmul:
            out = out.view(*original_shape[:-1], out.size(-1))

        if bias is not None:
             out += bias

        return out
```
FYI, I can run glm-4v inferenve with this change. :)